src/core/algorithms/gfd/gfd_miner.cpp

```diff
@@ -0,0 +1,648 @@
+                    for (; adjacency_it != adjacency_end; ++adjacency_it) {
+                        boost::graph_traits<graph_t>::adjacency_iterator pat_adjacency_it,
+                                pat_adjacency_end;
+                        boost::tie(pat_adjacency_it, pat_adjacency_end) =
+                                boost::adjacent_vertices(u, pattern);
+                        bool matched = false;
+                        for (; pat_adjacency_it != pat_adjacency_end; ++pat_adjacency_it) {
+                            if (embedding.at(*pat_adjacency_it) == *adjacency_it) {
+                                matched = true;
+                                break;
+                            }
+                        }
+                        if (matched ||
+                            graph[*adjacency_it].attributes.at("label") != vertex_label ||
+                            graph[boost::edge(v, *adjacency_it, graph).first].label != edge_label) {
+                            continue;
+                        }
+                        Embedding new_embedding(std::move(embedding));
+                        new_embedding.emplace(new_vertex, std::move(*adjacency_it));
+                        new_embeddings.push_back(std::move(new_embedding));
+                    }
+                }
+                if (!new_embeddings.empty()) {
+                    new_forbidden_rules_set.push_back(forbidden_rules);
+                    new_patterns.push_back(std::move(new_pattern));
+                    new_embeddings_set.push_back(std::move(new_embeddings));
+                }
+            }
+        }
+    }
+}
+
+}  // namespace
+
+namespace algos {
+
+GfdMiner::GfdMiner() : Algorithm({}) {
+    RegisterOptions();
+    MakeOptionsAvailable(
+            {config::names::kGraphData, config::names::kGfdK, config::names::kGfdSigma});
+}
+
+void GfdMiner::RegisterOptions() {
+    DESBORDANTE_OPTION_USING;
+
+    RegisterOption(config::Option{&graph_path_, kGraphData, kDGraphData});
+    RegisterOption(config::Option{&k_, kGfdK, kDGfdK});
+    RegisterOption(config::Option{&sigma_, kGfdSigma, kDGfdSigma});
+}
+
+void GfdMiner::LoadDataInternal() {
+    std::ifstream f(graph_path_);
+    graph_ = parser::graph_parser::ReadGraph(f);
+}
+
+void GfdMiner::ResetState() {}
+
+unsigned long long GfdMiner::ExecuteInternal() {
+    std::size_t elapsed_time = util::TimedInvoke(&GfdMiner::MineGfds, this, graph_, k_, sigma_);
+    LOG(DEBUG) << "Mined GFDs: " << gfds_.size();
+    return elapsed_time;
+}
+
+void GfdMiner::MineGfds(graph_t const& graph, std::size_t const k, std::size_t const sigma) {
+    std::set<std::string> vertex_labels = {};
+    std::set<std::string> edge_labels = {};
+    std::vector<graph_t> patterns = {};
+
+    std::vector<Embeddings> embeddings_set = {};
+    std::vector<Rules> forbidden_rules_set = {};
+    std::map<std::string, Info> attrs_info = {};
```

Comment:
You should consider making some of these variables class fields in order to decrease the amount passed arguments to the functions below.
Also make those functions class methods.